### DISCUSSION TITLE
Textarea in JSON Schema Bridge

### DISCUSSION THREAD

**Discussion by kevinsimper** [vazco/uniforms#979] [Q&A] [answered]
We are using JSON Schema bridge and we are starting up the form with a JSON provided from the backend.

I looked into the source and it looks like there is no way to set that a string field needs to be a textarea.

```
{
  "title": "Address",
  "type": "object",
  "properties": {
    "city": { "type": "string" },
    "state": { "type": "string" },
    "street": { "type": "string", "uniforms": { "type": "textarea" } },
    "zip": { "type": "string", "pattern": "[0-9]{5}" },
  },
  "required": ["street", "zip", "state"],
}
```

**Comment by wadamek65** [comment, marked as answer]
Hi @kevinsimper ,

Yes you're right, there is no way to tell uniforms to render a textarea unless the theme accepts a prop which transforms the `TextField` into one as that is the one that by default is rendered for all strings (for example `multiline: true` should work for material theme).

What you probably want to do here is to tell uniforms to render this field as a `LongTextField` by passing it in `uniforms.component` property like this:
```
{
  "title": "Address",
  "type": "object",
  "properties": {
    "city": { "type": "string" },
    "state": { "type": "string" },
    "street": { "type": "string", "uniforms": { "component": LongTextField } },
    "zip": { "type": "string", "pattern": "[0-9]{5}" },
  },
  "required": ["street", "zip", "state"],
}
```

Since you are fetching the schema from the server, you will have to derive it on client-side like for example explained [here](https://github.com/vazco/uniforms/issues/973#issuecomment-859520556).
